flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -404,15 +404,16 @@ private void stopResourceManagerServices() throws Exception {
 		final WorkerRegistration<WorkerType> workerTypeWorkerRegistration = taskExecutors.get(taskManagerResourceId);
 
 		if (workerTypeWorkerRegistration.getInstanceID().equals(taskManagerRegistrationId)) {
-			registerTaskManagerAtSlotManager(workerTypeWorkerRegistration, slotReport);
+			slotManager.registerTaskManager(workerTypeWorkerRegistration, slotReport);
+			onTaskManagerRegistration(workerTypeWorkerRegistration);
```

Comment:
I think we might call `onTaskManagerRegistration` multiple times if the `TM` decides to call `sendSlotReport` repeatedly.